stubs/tensorflow/tensorflow/keras/metrics.pyi

```diff
@@ -1,9 +1,136 @@
-from tensorflow import Tensor
-from tensorflow._aliases import TensorCompatible
+from _typeshed import Incomplete
+from abc import ABCMeta, abstractmethod
+from collections.abc import Callable, Iterable, Sequence
+from typing import Any, Literal
+from typing_extensions import Self, TypeAlias, override
 
+import tensorflow as tf
+from tensorflow import Operation, Tensor
+from tensorflow._aliases import DTypeLike, KerasSerializable, TensorCompatible
+from tensorflow.keras.initializers import _Initializer
+from tensorflow.keras.layers import _Constraint
+from tensorflow.keras.regularizers import _Regularizer
+
+_Output: TypeAlias = Tensor | dict[str, Tensor]
+
+class Metric(tf.keras.layers.Layer[tf.Tensor, tf.Tensor], metaclass=ABCMeta):
+    def __init__(self, name: str | None = None, dtype: DTypeLike | None = None) -> None: ...
+    def __new__(cls, *args: Any, **kwargs: Any) -> Self: ...
+    def __call__(self, *args: Incomplete, **kwargs: Incomplete) -> Incomplete: ...
+    def __deepcopy__(self, memo: Incomplete | None = None) -> Incomplete: ...
+    @property
+    def _trackable_saved_model_saver(self) -> Incomplete: ...
+    @property
+    def trainable_weights(self) -> list[Incomplete]: ...
+    @property
+    def non_trainable_weights(self) -> Incomplete: ...
+    def merge_state(self, metrics: Iterable[Self]) -> list[Operation]: ...
+    def reset_state(self) -> None: ...
+    @abstractmethod
+    def update_state(
+        self, y_true: TensorCompatible, y_pred: TensorCompatible, sample_weight: TensorCompatible | None = None
+    ) -> Operation | None: ...
+    @abstractmethod
+    def result(self) -> _Output: ...
+    @classmethod
+    def from_config(cls, config: dict[str, Any]) -> Self: ...
+    def get_config(self) -> dict[str, Any]: ...
+    @override
+    def add_weight(
```

Comment:
This also looks like an inherited method so metrics class shouldn't have it defined.

Reply:
Done in e9246aa7b. It was already inherited from Layer indeed.

Reply:
@hmc-cs-mdrissi The `add_weight` from `keras.metrics.Metric` is different from the one from `keras.layers.Layer`, so I need to redefine it. However I'm facing the same error I had in another PR with `MultiHeadAttention`. Even if I add the method to the `stubtest_allowlist`, mypy still reports an error.

Reply:
`# type: ignore` works for that. stubtest allowlist is for runtime vs stub checks. Before that happens there's also a mypy pass. And you can use normal  `# type: ignore` to handle that.

Reply:
Generally get mypy passing first. No need to consider allowlist and then after mypy is green work on stubtest errors, and use allowlist when appropriate.

Reply:
Ok, thanks.